### PR TITLE
Better error printing

### PR DESCRIPTION
**Pull request by granjef3**
The default `Termination` impl for `Result` uses `Debug` to print any `Result::Err` returned by `main`. This PR overrides that behavior by moving the contents of main into a wrapper function that we can print the result of ourselves using `Display` instead.
This will print errors in a more human readable fashion.

### CODE REVIEW

File: src/terminal/emoji.rs [hunk 1]
```diff
@@ -25,6 +25,6 @@ pub static SLEUTH: Emoji = Emoji("🕵️‍♂️", "");
 pub static SPARKLES: Emoji = Emoji("✨  ", "");
 pub static SWIRL: Emoji = Emoji("🌀 ", "");
 pub static UP: Emoji = Emoji("🆙 ", "");
-pub static WARN: Emoji = Emoji("⛔ ", "");
+pub static WARN: Emoji = Emoji("⚠️", "");
```
Comment (EverlastingBugstopper) [inline src/terminal/emoji.rs]: change to ⚠

Comment (granjef3) [reply src/terminal/emoji.rs]: it is, github can't unicode

**Comment by granjef3** [comment]
https://github.com/tismith/exitfailure might be a better solution

File: src/terminal/emoji.rs (hunk 1 above)
Comment (EverlastingBugstopper) [inline src/terminal/emoji.rs]: You need to add a space after the emoji here to match the style of the rest. After that, this looks good to me!

Comment (granjef3) [inline src/terminal/emoji.rs]:
```suggestion
pub static WARN: Emoji = Emoji("⚠️ ", "");
```

**Pull request merged**